entrypoint.sh

```diff
@@ -70,10 +71,7 @@ case "${ACTION}" in
     ;;
 esac
 
-echo "${OUTPUT}"
-SINGLE_LINE_OUTPUT="${OUTPUT//'%'/'%25'}"
-SINGLE_LINE_OUTPUT="${SINGLE_LINE_OUTPUT//$'\n'/'%0A'}"
-SINGLE_LINE_OUTPUT="${SINGLE_LINE_OUTPUT//$'\r'/'%0D'}"
+SINGLE_LINE_OUTPUT=$(echo "${OUTPUT}" | awk 'BEGIN {RS=""}{gsub(/%/,"%25") gsub(/\r/, "%0D") gsub(/\n/, "%0A")}1')
```

Comment:
I'd do the `print` bit to be less cryptic (it's equivalent to what you had).

The rest of the suggestions are maybe my personal preference, so feel free to ignore them if you think it's more ugly (makes it more readable to me).

```suggestion
SINGLE_LINE_OUTPUT=$(echo "${OUTPUT}" | awk 'BEGIN { RS="" } { gsub(/%/,"%25"); gsub(/\r/, "%0D"); gsub(/\n/, "%0A") } { print }')
```